### PR TITLE
Feature/add rest endpoint

### PR DESCRIPTION
**Pull request by SpiderDan98**
Add one-time-password rest endpoint

### CODE REVIEW

File: bundles/one-time-password-rest-api/LICENSE [hunk 1]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Comment (sourcecube) [inline bundles/one-time-password-rest-api/LICENSE:3]: 2021 (;

File: bundles/one-time-password-rest-api/composer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
+    "spryker/glue-application": "^1.21.0"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "dev-master as 1.0.0",
```
Comment (sourcecube) [inline bundles/one-time-password-rest-api/composer.json:17]: You can use 1.0.0, it was released 8 days ago (:

File: bundles/one-time-password-rest-api/src/FondOfOryx/Shared/OneTimePasswordRestApi/Transfer/one_time_password_rest_api.transfer.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<transfers xmlns="spryker:transfer-01" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="spryker:transfer-01 http://static.spryker.com/transfer-01.xsd">
+    <transfer name="RestOneTimePasswordRequestAttributes">
+        <property name="email" type="string"/>
+    </transfer>
+
+    <transfer name="OneTimePasswordResponse">
```
Comment (sourcecube) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Shared/OneTimePasswordRestApi/Transfer/one_time_password_rest_api.transfer.xml:8]: Should this be namespaced with `Rest` as well like the request? Or ist that how it's done in Spyker core?

File: bundles/one-time-password/LICENSE [hunk 4]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Comment (sourcecube) [inline bundles/one-time-password/LICENSE:3]: 2021.. and for future record, these files should be in a separate PR

File: bundles/one-time-password/composer.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      "name": "Daniel Kemna",
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "dev-master as 1.0.0",
```
Comment (sourcecube) [inline bundles/one-time-password/composer.json:16]: `^1.0`

File: bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/OneTimePasswordRestApiConfig.php [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+use Spryker\Glue\Kernel\AbstractBundleConfig;
+
+class OneTimePasswordRestApiConfig extends AbstractBundleConfig
+{
+    public const RESOURCE_ONE_TIME_PASSWORD = 'one-time-password';
+    public const CONTROLLER_ONE_TIME_PASSWORD = 'one-time-password-resource';
+
+    public const EMAIL_REQUIRED_ERROR_CODE = '1000';
+    public const EMAIL_REQUIRED_ERROR_DETAIL = 'The attribute Email is required.';
```
Comment (sourcecube) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/OneTimePasswordRestApiConfig.php:13]: Nice! Maybe add a comment like .. "The code is not related to a system yet, might change later"

File: bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        RestOneTimePasswordRequestAttributesTransfer $restOneTimePasswordRequestAttributesTransfer
+    ): RestResponseInterface {
+        if (!$restOneTimePasswordRequestAttributesTransfer->getEmail()) {
+            return $this->createEmailRequiredError();
+        }
+
+        return $this->restResourceBuilder
+            ->createRestResponse()
+            ->setStatus(204);
+    }
```
Comment (sourcecube) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:44]: You could add a test for this to make sure that the error will be returned

**Review by sourcecube — APPROVED** [review]

**Comment by sourcecube** [comment]
Good work 🎉

File: bundles/one-time-password-rest-api/composer.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Daniel Kemna",
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
+    "spryker/glue-application": "^1.21.0"
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:13]: Please use ^1.0.0

File: bundles/one-time-password-rest-api/composer.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  "description": "Spryker One Time Password Rest Api Module.",
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Daniel Kemna",
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:12]: 7.2 EOL

File: composer.json [hunk 10]
```diff
@@ -19,6 +19,7 @@
     "spryker/customer": "^6.0.0 || ^7.0.0",
     "spryker/calculation": "^4.4.1",
     "spryker/country": "^3.0.0",
+    "spryker/glue-application": "^1.21.0",
```
Comment (daniel-rose) [inline composer.json:22]: Use ^1.0.0

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: bundles/one-time-password-rest-api/composer.json [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.3",
+    "spryker/glue-application": "^1.0.0"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "^1.0.0",
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:17]: use only `"fond-of-codeception/spryker": "^1.0",`

File: bundles/one-time-password-rest-api/composer.json [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    "mikey179/vfsstream": "^1.6",
+    "phpro/grumphp": "^0.14",
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:19]: remove unused stuff

File: bundles/one-time-password-rest-api/composer.json [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    "sebastian/phpcpd": "^4.0",
+    "spryker/code-sniffer": "^0.11",
+    "spryker/development": "^3.6"
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:22]: Why?

File: bundles/one-time-password-rest-api/composer.json [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+  "require": {
+    "php": ">=7.3",
+    "spryker/glue-application": "^1.0.0"
+  },
+  "require-dev": {
+    "codeception/codeception": "^2.3",
+    "fond-of-codeception/spryker": "^1.0.0",
+    "mikey179/vfsstream": "^1.6",
+    "phpro/grumphp": "^0.14",
+    "sebastian/phpcpd": "^4.0",
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:20]: Why?

File: bundles/one-time-password-rest-api/composer.json [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/composer.json:34]: `"Generated\\": "src/Generated/"`

File: bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiDependencyProvider.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * @param \Spryker\Client\Kernel\Container $container
+     *
+     * @return \Spryker\Client\Kernel\Container
+     */
+    protected function addZedRequestClient(Container $container): Container
+    {
+        $container->set(
+            static::CLIENT_ZED_REQUEST,
+            function (Container $container) {
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiDependencyProvider.php:35]: use static closure please.. (performance)

File: bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiFactory.php [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+     */
+    public function createOneTimePasswordRestApiZedStub(): OneTimePasswordRestApiStubInterface
+    {
+        return new OneTimePasswordRestApiStub($this->getZedRequestClient());
+    }
+
+    /**
+     * @return \FondOfOryx\Client\OneTimePasswordRestApi\Dependency\Client\OneTimePasswordRestApiToZedRequestClientInterface
+     */
+    public function getZedRequestClient(): OneTimePasswordRestApiToZedRequestClientInterface
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/OneTimePasswordRestApiFactory.php:23]: Can be protected...

File: bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/Zed/OneTimePasswordRestApiStub.php [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+     * @param \Generated\Shared\Transfer\RestOneTimePasswordRequestAttributesTransfer $restOneTimePasswordRequestAttributesTransfer
+     *
+     * @return \Generated\Shared\Transfer\RestOneTimePasswordResponseTransfer
+     */
+    public function requestOneTimePassword(
+        RestOneTimePasswordRequestAttributesTransfer $restOneTimePasswordRequestAttributesTransfer
+    ): RestOneTimePasswordResponseTransfer {
+        /**
+ * @var \Generated\Shared\Transfer\RestOneTimePasswordResponseTransfer $restOneTimePasswordResponseTransfer
+*/
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Client/OneTimePasswordRestApi/Zed/OneTimePasswordRestApiStub.php:34]: CodeStyle...

File: bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Controller/OneTimePasswordResourceController.php [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+use Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface;
+use Spryker\Glue\Kernel\Controller\AbstractController;
+
+/**
+ * @method \FondOfOryx\Glue\OneTimePasswordRestApi\OneTimePasswordRestApiFactory getFactory()
+ */
+class OneTimePasswordResourceController extends AbstractController
+{
+    /**
+     * @param \Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface $restRequest
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Controller/OneTimePasswordResourceController.php:16]: Add glue annotation please (for auto doc) --> https://documentation.spryker.com/docs/documenting-glue-api-resources#rest-methods

File: bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+            return $this->createEmailRequiredError();
+        }
+
+        $RestOneTimePasswordResponseTransfer = $this->oneTimePasswordClient->requestOneTimePassword(
+            $restOneTimePasswordRequestAttributesTransfer
+        );
+
+        return $this->restResourceBuilder
+            ->createRestResponse()
+            ->setStatus(204);
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:57]: Use const for this. (HttpFoundation from Symfony)

File: bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * @return \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResponseInterface
+     */
+    protected function createEmailRequiredError(): RestResponseInterface
+    {
+        $restErrorMessageTransfer = new RestErrorMessageTransfer();
+
+        $restErrorMessageTransfer->setStatus(400)
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/src/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessor.php:67]: Same

File: bundles/one-time-password-rest-api/tests/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessorTest.php [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+            ->with($this->restOneTimePasswordRequestAttributesTransferMock)
+            ->willReturn($this->restOneTimePasswordResponseTransferMock);
+
+        $this->restResourceBuilderMock->expects($this->atLeastOnce())
+            ->method('createRestResponse')
+            ->willReturn($this->restResponseMock);
+
+        $this->restResponseMock->expects($this->atLeastOnce())
+            ->method('setStatus')
+            ->with(204)
```
Comment (daniel-rose) [inline bundles/one-time-password-rest-api/tests/FondOfOryx/Glue/OneTimePasswordRestApi/Processor/OneTimePasswordProcessorTest.php:112]: Const from HttpFoundation

File: bundles/one-time-password/composer.json [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+  "description": "Spryker One Time Password Module.",
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Daniel Kemna",
+      "email": "danielkemna@t-online.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2"
```
Comment (daniel-rose) [inline bundles/one-time-password/composer.json:12]: EOL for 7.2. Please use >= 7.3...

File: bundles/one-time-password/composer.json [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Comment (daniel-rose) [inline bundles/one-time-password/composer.json:33]: Look at composer.json from one-time-password-rest-api...

**Review by daniel-rose — APPROVED** [review]
Keep it mind at dandelion.json

File: bundles/one-time-password-rest-api/composer.json (hunk 15 above)
Comment (julianzimmermann) [inline bundles/one-time-password-rest-api/composer.json:34]: src/Generated/

**Review by julianzimmermann — APPROVED** [review]

**Pull request merged**